partiql-parser/src/main/kotlin/org/partiql/parser/impl/PartiQLParserDefault.kt

```diff
@@ -1930,9 +1932,31 @@ internal class PartiQLParserDefault : PartiQLParser {
         }
 
         override fun visitLiteralInteger(ctx: GeneratedParser.LiteralIntegerContext) = translate(ctx) {
-            val n = ctx.LITERAL_INTEGER().text.toInt()
-            val v = intValue(n.toBigInteger())
-            exprLit(v)
+            val n = ctx.LITERAL_INTEGER().text
+
+            // 1st, try parse as int
```

Comment:
Leaving this as an optional change. If we aren't going to do anything with the exceptions:

```kotlin
val text = ctx.LITERAL_INTEGER().text
val integer = text.toIntOrNull()?.let { int32Value(it) }
    ?: text.toLongOrNull()?.let { int64Value(it) }
    ?: text.toBigIntegerOrNull()?.let { intValue(it) }
    ?: throw error(ctx, "Unable to parse integer: $text.")
exprLit(integer)
```